qualtran/drawing/flame_graph.py

```diff
@@ -0,0 +1,132 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Classes for drawing bloqs with FlameGraph."""
+import functools
+import pathlib
+import subprocess
+import tempfile
+from typing import Any, List, Optional, Sequence, Union
+
+import networkx as nx
+import numpy as np
+
+from qualtran import Bloq
+from qualtran.resource_counting.bloq_counts import _compute_sigma
+from qualtran.resource_counting.t_counts_from_sigma import t_counts_from_sigma
+
+
+def _pretty_arg(val: Any) -> str:
+    if isinstance(val, (tuple, np.ndarray)):
+        return f'{val.shape if isinstance(val, np.ndarray) else len(val)}'
+    if isinstance(val, Bloq):
+        return _pretty_name(val)
+    if isinstance(val, float):
+        if np.isclose(val, 0):
+            val = 0
+        return f'{val:0.2g}'
+    return f'{val}'
+
+
+def _pretty_name(bloq: Bloq) -> str:
+    from qualtran.serialization.bloq import _iter_fields
+
+    ret = bloq.pretty_name()
+    if bloq.pretty_name.__qualname__.startswith('Bloq.'):
+        for field in _iter_fields(bloq):
+            ret += f'[{_pretty_arg(getattr(bloq, field.name))}]'
+    return ret
+
+
+@functools.lru_cache(maxsize=None)
+def _t_counts_for_bloq(bloq: Bloq, graph: nx.DiGraph) -> int:
+    sigma = _compute_sigma(bloq, graph)
+    return t_counts_from_sigma(sigma)
```

Comment:
why not `t_complexity(bloq).t`?

Reply:
It doesn't sum up costs for rotation bloqs anymore, since we moved `ceil(1.149 * log2(1.0 / eps) + 9.2)` to `TComplexity.rotation_cost` and but we don't track the `eps` in the `TComplexity` object.